### ISSUE TITLE
🚀Support for Angular 12

### ISSUE THREAD

**Issue by kristofdc** [ngxs/store#1748]
### Relevant Package
@ngxs/store 

### Description
support for Angular v12 already available?

**Comment by markwhitfeld** [comment]
The current version should support Angular v12 (although the peer dependencies will complain).
I am currently in discussions with the angular team(and have been for the past month) on how we can include the new angular library format so that Angular 12 projects can take advantage of this, but we would also want to keep compatibility with older versions of Angular too if at all possible. It looks like my latest idea may just work! They have asked me to build a small prototype.

**Comment by cybercussion** [comment]
I'm getting -

```
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/core@">=6.1.0 <12.0.0" from @ngxs/logger-plugin@3.7.1
npm ERR! node_modules/@ngxs/logger-plugin
npm ERR!   @ngxs/logger-plugin@"^3.7.1" from the root project
```

**Comment by markwhitfeld** [comment]
@kristofdc @cybercussion Support for Angular 12 has just been officially released in NGXS v3.7.2
Please try it out and let me know if you have any issues. You are welcome to message me directly on the NGXS slack channel.

cc @danielsogl @perzeuss @kuncevic @bboehm86 @pkim

**Comment by david-sharer** [comment]
After updating we are having some issue with Angular 12 integration. Having a hard time tracing it.

```
Error: This constructor is not compatible with Angular Dependency Injection because its dependency at index 2 of the parameter list is invalid.
This can happen if the dependency type is a primitive like a string or if an ancestor of this class is missing an Angular decorator.

Please check that 1) the type for the parameter at index 2 is correct and 2) the correct Angular decorators are defined for this class and its ancestors.
    at ɵɵinvalidFactoryDep (core.js:4753)
    at Object.UserState_Factory [as factory] (ɵfac.js? [sm]:1)
    at R3Injector.hydrate (core.js:11412)
    at R3Injector.get (core.js:11232)
    at StateFactory.add (ngxs-store.js:2525)
    at StateFactory.addAndReturnDefaults (ngxs-store.js:2547)
    at new NgxsRootModule (ngxs-store.js:3433)
    at Object.NgxsRootModule_Factory [as factory] (ngxs-store.js:3441)
    at R3Injector.hydrate (core.js:11412)
    at R3Injector.get (core.js:11232)
 ```
 
 Going to follow up on the slack.
